### DISCUSSION TITLE
How to get disk.rm to work?

### DISCUSSION THREAD

**Discussion by MrCyrus** [vmware/govmomi#3280] [Community Help] [answered]
I'm working on automating an OS upgrade. I do this by using govc to create a temp disk to a VM to be used by the upgrade and when the upgrade is completed I need to now remove the disk but I can't find a way to get this to work. 

I've run:
    govc device.info -json -vm $vm 'disk-*'
to get all disks assigned to the $vm, and I then try to run:
    govc disk.rm -ds $ds $contentId
and this fails with "The object or item referred to could not be found". I even tried using $uuid as the ID and the error is the same. So would appreciate some help in figuring out how to get disk.rm to work.
Thank you.

**Comment by dougm** [comment, marked as answer]
Sorry for the delayed response. disk.rm is for First Class Disks, i.e. created with disk.create or disk.register. 
Sounds like you need to use [device.remove](https://github.com/vmware/govmomi/blob/main/govc/USAGE.md#deviceremove) here.
